sources/packages/backend/apps/queue-consumers/src/processors/schedulers/base-scheduler.ts

```diff
@@ -0,0 +1,40 @@
+import { OnApplicationBootstrap } from "@nestjs/common";
+import Bull, { CronRepeatOptions, Queue } from "bull";
+
+export abstract class BaseScheduler<T> implements OnApplicationBootstrap {
+  abstract initializeScheduler(): void;
```

Comment:
Please add comments also here.
It can start with something like `When overridden in a derived class....`.